src/megolm/group_session.rs

```diff
@@ -131,86 +127,34 @@ impl GroupSession {
     /// Convert the group session into a struct which implements
     /// [`serde::Serialize`] and [`serde::Deserialize`].
     pub fn pickle(&self) -> GroupSessionPickle {
-        GroupSessionPickle {
-            ratchet: self.ratchet.clone().into(),
-            signing_key: self.signing_key.clone(),
-        }
+        GroupSessionPickle { ratchet: self.ratchet.clone(), signing_key: self.signing_key.clone() }
     }
 
-    /// Pickle the group session and serialize it to a JSON string.
-    ///
-    /// The string is wrapped in [`GroupSessionPickledJSON`] which can be
-    /// derefed to access the content as a string slice. The string will zeroize
-    /// itself when it drops to prevent secrets contained inside from lingering
-    /// in memory.
-    ///
-    /// [`GroupSessionPickledJSON`]: self::GroupSessionPickledJSON
-    pub fn pickle_to_json_string(&self) -> GroupSessionPickledJSON {
-        let pickle: GroupSessionPickle = self.pickle();
-        GroupSessionPickledJSON(
-            serde_json::to_string_pretty(&pickle).expect("Group session serialization failed."),
-        )
+    pub fn from_pickle(pickle: GroupSessionPickle) -> Self {
+        pickle.into()
     }
 }
 
+/// A format suitable for serialization which implements [`serde::Serialize`]
+/// and [`serde::Deserialize`]. Obtainable by calling [`GroupSession::pickle`].
 #[derive(Serialize, Deserialize)]
 pub struct GroupSessionPickle {
-    ratchet: RatchetPickle,
+    ratchet: Ratchet,
     signing_key: Ed25519Keypair,
 }
 
-/// A format suitable for serialization which implements [`serde::Serialize`]
-/// and [`serde::Deserialize`]. Obtainable by calling [`GroupSession::pickle`].
 impl GroupSessionPickle {
-    /// Convert the pickle format back into a [`GroupSession`].
-    pub fn unpickle(self) -> Result<GroupSession, GroupSessionUnpicklingError> {
-        self.try_into()
-    }
-}
-
-impl TryFrom<GroupSessionPickle> for GroupSession {
-    type Error = GroupSessionUnpicklingError;
-
-    fn try_from(pickle: GroupSessionPickle) -> Result<Self, Self::Error> {
-        Ok(Self { ratchet: pickle.ratchet.try_into()?, signing_key: pickle.signing_key })
-    }
-}
-
-#[derive(Zeroize, Debug)]
-#[zeroize(drop)]
-pub struct GroupSessionPickledJSON(String);
```

Comment:
The point of this type was to avoid people serializing to `String` which would then be left unzeroized since `String` doesn't implement zeroization on drop. Instead, this wrapper type offered to handle this for you while providing a way to access the raw serialized content via borrowing.

However, from internal discussions with @poljar, we're worried about the proliferation of types being hard to explain and hard to use. Instead, we're considering placing prominent warnings in the documentation that users of pickling functionality should take care to zeroize their serialization buffers themselves or use the `encrypt` method.

Can we please add a TODO to add such docs?

Reply:
Opened an issue https://github.com/matrix-org/vodozemac/issues/45